src/core/lib/surface/init.cc

```diff
@@ -208,6 +211,8 @@ void grpc_shutdown(void) {
     grpc_core::ApplicationCallbackExecCtx* acec =
         grpc_core::ApplicationCallbackExecCtx::Get();
     if (!grpc_iomgr_is_any_background_poller_thread() &&
+        !grpc_event_engine::posix_engine::TimerManager::
```

Comment:
Is it possible a TimerManager thread is holding the last initialization ref? I think we could add a grpc_init/grpc_shutdown pair in TimerManager constructor/destructor respectively to ensure that isn't the case.

Reply:
That would tie grpc initialization to timer manager and create a dependency loop.

Reply:
That's rough. I think the scenario is possible, it's easy to construct artificially. And I think we'll have some immediate problems if the internal shutdown is never called (Asylo will be a good canary there, if nothing else). Can we add a test where a timer callback holds the last init ref? I laid out the basic framework for TimerManager tests here https://github.com/grpc/grpc/pull/30758

Reply:
done... I'm going to import this and run it against asylo too

Reply:
asylo looks good